lib/api/apiUtils/object/BackendInfo.js

```diff
@@ -63,6 +71,42 @@ export class BackendInfo {
     }
 
     /**
+     * validate that s3 server is running with a file or memory backend
+     * @param {string} requestEndpoint - request endpoint
+     * @param {object} log - werelogs logger
+     * @return {boolean} - true if running with file/mem backend, false if not
+     */
+    static isMemOrFileBackend(requestEndpoint, log) {
+        if (config.backends.data === 'mem' ||
+        config.backends.data === 'file') {
+            log.trace('use data backend for the location', {
+                dataBackend: config.backends.data,
+                method: 'isMemOrFileBackend',
+            });
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * validate requestEndpoint against config or mem/file data backend
+     * - if there is a mismatch between the request endpoint and what is in
```

Comment:
I would change this to something like "if there is no match for the request endpoint in the config restEndpoints....."
just to make it a bit clearer